library/core/src/mem/maybe_uninit.rs

```diff
@@ -910,52 +910,6 @@ impl<T> MaybeUninit<T> {
         }
     }
 
-    /// Extracts the values from an array of `MaybeUninit` containers.
-    ///
-    /// # Safety
-    ///
-    /// It is up to the caller to guarantee that all elements of the array are
-    /// in an initialized state.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// #![feature(maybe_uninit_uninit_array)]
-    /// #![feature(maybe_uninit_array_assume_init)]
-    /// use std::mem::MaybeUninit;
-    ///
-    /// let mut array: [MaybeUninit<i32>; 3] = MaybeUninit::uninit_array();
-    /// array[0].write(0);
-    /// array[1].write(1);
-    /// array[2].write(2);
-    ///
-    /// // SAFETY: Now safe as we initialised all elements
-    /// let array = unsafe {
-    ///     MaybeUninit::array_assume_init(array)
-    /// };
-    ///
-    /// assert_eq!(array, [0, 1, 2]);
-    /// ```
-    #[unstable(feature = "maybe_uninit_array_assume_init", issue = "96097")]
-    #[rustc_const_unstable(feature = "const_maybe_uninit_array_assume_init", issue = "96097")]
-    #[inline(always)]
-    #[track_caller]
-    pub const unsafe fn array_assume_init<const N: usize>(array: [Self; N]) -> [T; N] {
```

Comment:
Like I've mentioned in some other PRs, I'm not willing to take an API removal (even an unstable one) without libs-api team oversight.

Up to you if you want to bring this up in an ACP to see if T-libs-api agrees, or if you want to make this just a T-libs change that updates the implementations of things to use `transpose` without actually removing things.

@rustbot author

Reply:
Nah, I want to get rid of these. Created https://github.com/rust-lang/libs-team/issues/122

Reply:
Actually, this is probably what you were trying to tell me but I think it'll be easier to split stdlib usage removal from api removal, so I'll pull out the usage removals.

Reply:
Yup, it's always easiest to split things up by decision makers.

Such as, where possible:
- changes to `compiler` separate from `library`
- additions separate from removals
- internal details separate from anything impacting callers

Reply:
Haha yeah, thanks. I feel like I've learned this lesson at least a dozen times by now, but the initial ease of piling a bunch of crap together is too good a fool. :)